### PR TITLE
Read session token directly from document cookie before making API calls

### PR DESCRIPTION
**Pull request by djvirgen**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -45,7 +45,14 @@
 
         // The user's cookies are required by the API to properly handle the request.
         // Keep a reference so that we can ensure we use updated cookies.
-        this._cookies = options.cookies || {};
+        this._cookies = {};
+        if (options.cookies) {
+            var cookies = options.cookies.split(/; */);
```
Comment (pinktrink) [inline lib/index.js]: Can we initialize this regex earlier so it's not being recompiled each time this hits?

Comment (djvirgen) [inline lib/index.js]: Good idea.

Comment (djvirgen) [inline lib/index.js]: Done!

File: lib/index.js [hunk 2]
```diff
@@ -89,8 +98,16 @@
             query[key] = this._options.query[key];
         }
 
-        if (this._cookies.S) {
-            query.session_token = this._cookies.S
+        var sessionToken;
+
+        if ('getSessionToken' in this._http) {
```
Comment (djvirgen) [inline lib/index.js]: Should I use `typeof this._http.getSessionToken === 'function'` instead?

Comment (pinktrink) [inline lib/index.js]: Hell, quickest would simply be `this._http.getSessionToken`

Comment (djvirgen) [inline lib/index.js]: Nah, maybe I'm just being too cautious. A simple "does it exist" check is probably fine.

**Pull request merged**